packages/immutadot-parser/src/parser.ts

```diff
@@ -0,0 +1,132 @@
+import { Maybe } from "./maybe"
+
+export type Parser<T> = (input: string) => T
+
+/**
+ * Creates a parser from a regular expression by matching the input string with
+ * the regular expression, returning the resulting match object.
+ *
+ * @param regexp the regular expression
+ * @returns the resulting parser
+ *
+ * @remarks
+ * Since 1.0.0
+ */
+export const fromRegExp = (regexp: RegExp): Parser<Maybe<string[]>> => (str) => Maybe.map(
+  str.match(regexp),
+  (match) => match.slice(1),
```

Comment:
Replace by an array destructuring discarding the first element, naming the rest `groups` ?